lib/logstash/inputs/snmptrap.rb

```diff
@@ -47,26 +47,44 @@ def register
     end
   end # def register
 
-  public
   def run(output_queue)
     begin
       # snmp trap server
       snmptrap_listener(output_queue)
     rescue => e
       @logger.warn("SNMP Trap listener died", :exception => e, :backtrace => e.backtrace)
-      sleep(5)
-      retry
+      stoppable_call(20) { sleep(0.25) }
```

Comment:
However the pipeline thread (code) does not know how the plugin thread came to be asleep or dead or whether the plugin implements the sleep protocol correctly (third party closed source plugins perhaps?), therefore, **is** it safer for the pipeline to check if the corresponding thread is dead or asleep before stopping the input or waking the corresponding thread? Do you need to stop an input if its thread is dead.  Our secondary threads die on unhandled exceptions e.g. OOM.

And we should summarize all the above is in the discussion issue that Colin is creating.